armbalancer.go

```diff
@@ -42,6 +44,13 @@ func New(opts Options) http.RoundTripper {
 	if opts.Host == "" {
 		opts.Host = "management.azure.com"
 	}
+	host, port, err := net.SplitHostPort(opts.Host)
```

Comment:
Please add a unit test to prove that this code behaves as expected when a port is specified, not specified, or some invalid host is provided.

Reply:
added some ut cases. PTAL. Thanks!